onnxruntime/core/providers/cuda/tensor/quantize_linear.cc

```diff
@@ -240,33 +506,60 @@ REGISTER_DQ_KERNEL_TYPED_10_12(uint8_t)
 REGISTER_DQ_KERNEL_TYPED_13_18(int8_t)
 REGISTER_DQ_KERNEL_TYPED_13_18(uint8_t)
 
-#define REGISTER_DQ_KERNEL_TYPED_19(T)                                     \
-  ONNX_OPERATOR_TWO_TYPED_KERNEL_EX(                                       \
+#define REGISTER_DQ_KERNEL_TYPED_19_20(T)                                  \
+  ONNX_OPERATOR_VERSIONED_TWO_TYPED_KERNEL_EX(                             \
       DequantizeLinear,                                                    \
       kOnnxDomain,                                                         \
-      19,                                                                  \
+      19, 20,                                                              \
       T, float,                                                            \
       kCudaExecutionProvider,                                              \
       (*KernelDefBuilder::Create())                                        \
           .TypeConstraint("T1", DataTypeImpl::GetTensorType<T>())          \
           .TypeConstraint("T2", DataTypeImpl::GetTensorType<float>()),     \
       DequantizeLinear<T, float>);                                         \
-  ONNX_OPERATOR_TWO_TYPED_KERNEL_EX(                                       \
+  ONNX_OPERATOR_VERSIONED_TWO_TYPED_KERNEL_EX(                             \
       DequantizeLinear,                                                    \
       kOnnxDomain,                                                         \
-      19,                                                                  \
+      19, 20,                                                              \
       T, MLFloat16,                                                        \
       kCudaExecutionProvider,                                              \
       (*KernelDefBuilder::Create())                                        \
           .TypeConstraint("T1", DataTypeImpl::GetTensorType<T>())          \
           .TypeConstraint("T2", DataTypeImpl::GetTensorType<MLFloat16>()), \
       DequantizeLinear<T, MLFloat16>);
 
-REGISTER_DQ_KERNEL_TYPED_19(int8_t)
-REGISTER_DQ_KERNEL_TYPED_19(uint8_t)
+REGISTER_DQ_KERNEL_TYPED_19_20(int8_t)
+REGISTER_DQ_KERNEL_TYPED_19_20(uint8_t)
+#if !defined(DISABLE_FLOAT8_TYPES)
+REGISTER_DQ_KERNEL_TYPED_19_20(Float8E4M3FN)
+REGISTER_DQ_KERNEL_TYPED_19_20(Float8E5M2)
+#endif
+
+#define REGISTER_DQ_KERNEL_TYPED_21(T, U)                          \
```

Comment:
>REGISTER_DQ_KERNEL_TYPED_21 [](http://example.com/codeflow?start=8&length=27)

nit: keep it consistent with REGISTER_DQ_KERNEL_TYPED_19_20, i.e., register T for type float and float16

Reply:
updated to new way